Neos.Media/Classes/Command/MediaCommandController.php

```diff
@@ -388,6 +409,161 @@ public function renderThumbnailsCommand(int $limit = null, bool $quiet = false)
         }
     }
 
+    /**
+     * Tidy up outdated imageVariants
+     *
+     * This command iterates over all existing asset variants with outdated preset.
+     * They can get filtered and deleted.
+     *
+     * @param string $assetSource If specified, only assets of this asset source are considered. For example "neos" or "my-asset-management-system".
+     * @param bool $quiet If set, only errors and questions will be displayed.
+     * @param bool $assumeYes If set, "yes" is assumed for the "shall I remove ..." dialog.
+     * @param bool $filterCroppedVariants Find custom crops but have to check if used (slow).
+     * @param int|null $limit Limit the result of unused assets displayed and removed for this run.
+     * @param string $presetIdentifier If specified, only presets with this identifier are observed.
+     */
+    public function removeOutdatedVariantsCommand(bool $quiet = false, bool $assumeYes = false, bool $filterCroppedVariants = false, int $limit = null, string $assetSource = '', string $presetIdentifier = '')
+    {
+        if (empty($presetIdentifier)) {
+            // will read all presets defined in 'Settings.Neos.Media.yaml'
+            $currentPresets = $this->imageVariantService->getAllPresetsByConfiguration();
+        } else {
+            // is --preset-identifier used
+            $currentPresets = $this->imageVariantService->getAllPresetsOfIdentifier($presetIdentifier);
+        }
+
+        if (empty($currentPresets)) {
+            !$quiet && $this->output->outputLine(PHP_EOL . PHP_EOL . '<em>No presets found.</em>');
+            exit;
```

Comment:
```suggestion
            exit(1);
```

Same as above. Maybe debatable here and I am fine either way as the user cannot know if presets exist or not, so this might be a valid early exit condition.